src/components/bottom_section/Buttons.js

```diff
@@ -0,0 +1,180 @@
+import React from 'react';
+
+import {
+  BLANK,
+  DIVISION,
+  MAX_CHARACTERS,
+  MINUS,
+  MULTIPLY,
+  PERCENTAGE,
+  PLUS,
+} from '../../constants/Constants';
+import {
+  AllClearButton,
+  CommaButton,
+  DivisionButton,
+  EqualsButton,
+  MinusButton,
+  MultiplyButton,
+  PercentageButton,
+  PlusButton,
+  SignButton,
+} from './buttons/index';
+
+const Buttons = ({
+  setDisplayValue,
+  displayValue,
+  waitingValue,
+  setWaitingValue,
+  setCurrentValue,
+  currentValue,
+  operation,
+  setOperation,
+}) => {
+  const currentValueAsNumber = Number(currentValue);
+  const displayValueAsNumber = Number(displayValue);
+  const displayValueAsString = String(displayValue);
+
+  const handleButtonClick = ({ target: { value } }) => {
+    getButtonValue(value);
+  };
+
+  const getButtonValue = value => {
+    if (displayValue === 0 || waitingValue) {
+      setDisplayValue(value);
+      setWaitingValue(false);
+    } else if (displayValueAsString.length < MAX_CHARACTERS)
+      setDisplayValue(displayValueAsString + value);
+  };
+
+  const checkResultLength = result => {
+    const resultAsString = String(result);
+    if (resultAsString.length > MAX_CHARACTERS) {
+      return resultAsString.slice(0, MAX_CHARACTERS - 1) + 'e';
+    } else return result;
+  };
+
+  const displayResult = () => {
+    switch (operation) {
+      case PLUS:
+        setDisplayValue(
+          checkResultLength(displayValueAsNumber + currentValueAsNumber)
+        );
+        setCurrentValue(displayValueAsNumber + currentValueAsNumber);
+        setWaitingValue(true);
+        setOperation(BLANK);
```

Comment:
You didn't fix the problem I mentioned before (doing same operation twice) + pls take a look closer at the switch cases: can't you extract them to a separate function???